gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -171,46 +166,272 @@ def _get_prior_measurements_in_world_frame(
+        w_i2Ui1_dict_tracks: RelativeDirectionsDict,
+    ) -> Tuple[RelativeDirectionsDict, RelativeDirectionsDict, Set[int]]:
+        """Perform inlier detection for the relative direction measurements.
+
+        Args:
+            w_i2Ui1_dict: Dictionary of Unit3 relative translations between cameras.
+            w_i2Ui1_dict_tracks: Dictionary of Unit3 relative translations between cameras and landmarks.
+
+        Returns:
+            Tuple of:
+            inlier_w_i2Ui1_dict: Dictionary of inlier Unit3 relative translations between cameras.
+            inlier_w_i2Ui1_dict_tracks: Dictionary of inlier Unit3 relative translations between cameras and landmarks.
+            inlier_cameras: Set of inlier cameras.
+        """
+
+        # Sample directions for projection
+        combined_measurements = list(w_i2Ui1_dict.values()) + list(w_i2Ui1_dict_tracks.values())
+        projection_directions = self.__sample_projection_directions(combined_measurements)
+
+        # Convert to measurements: map indexes to symbols.
+        dummy_noise_model = gtsam.noiseModel.Isotropic.Sigma(3, 1e-2)  # MFAS does not use this.
+        w_i1Ui2_measurements = self._binary_measurements_from_dict(w_i2Ui1_dict, w_i2Ui1_dict_tracks, dummy_noise_model)
 
-    def __get_initial_values(self, wTi_initial: List[Optional[PosePrior]]):
+        # Compute outlier weights using MFAS.
+        # TODO(ayush): parallelize this step.
+        outlier_weights: List[Dict[Tuple[int, int], float]] = []
+        for direction in projection_directions:
+            mfas_instance = MFAS(w_i1Ui2_measurements, direction)
+            outlier_weights.append(mfas_instance.computeOutlierWeights())
+        logger.debug("Computed outlier weights using MFAS.")
+
+        # Compute average outlier weight.
+        outlier_weights_sum: DefaultDict[Tuple[int, int], float] = defaultdict(float)
+        inliers = set()
+        for outlier_weight_dict in outlier_weights:
+            for w_i1Ui2 in w_i1Ui2_measurements:
+                i1, i2 = w_i1Ui2.key1(), w_i1Ui2.key2()
+                outlier_weights_sum[(i1, i2)] += outlier_weight_dict[(i1, i2)]
+        for (i1, i2), weight_sum in outlier_weights_sum.items():
+            if weight_sum / len(projection_directions) < OUTLIER_WEIGHT_THRESHOLD:
+                inliers.add((i1, i2))
+
+        # Filter outliers, index back from symbol to int.
+        # `inliers` contains both camera-camera and camera-landmark inliers. We separate them here.
+        inlier_w_i2Ui1_dict = {}
+        inlier_w_i2Ui1_dict_tracks = {}
+        inlier_cameras: Set[int] = set()
+        for (i1, i2) in w_i2Ui1_dict:
+            if (C(i2), C(i1)) in inliers:  # there is a flip in indices from w_i2Ui1_dict to inliers.
+                inlier_w_i2Ui1_dict[(i1, i2)] = w_i2Ui1_dict[(i1, i2)]
+                inlier_cameras.add(i1)
+                inlier_cameras.add(i2)
+
+        for (track_id, cam_id) in w_i2Ui1_dict_tracks:
```

Comment:
optional: we could also try something like
```python
for (j, i) in w_jUi_dict_tracks:
    if (C(i), L(j)) in inliers and i in inlier_cameras:
        inlier_w_jUi_dict_tracks[(j, i)] = w_jUi_dict_tracks[(j, i)]
```
to keep the `i` (camera index) and `j` (landmark index) notation we use elsewhere